lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,274 @@
+defmodule Thrift.Generator.Client do
+  defmodule Framed do
+    alias Thrift.Generator.Service
+    alias Thrift.Parser.Models.Function
+
+    def generate(service_module, service) do
+      functions = service.functions
+      |> Map.values
+      |> Enum.map(&generate_function(service_module, &1))
+
+      quote do
```

Comment:
It seems to be generating a lot more code than required. The Connection process is the same for all framed clients. Why use macros for this?